src/main/java/core/basesyntax/strategy/OperationHandler.java

```diff
@@ -0,0 +1,8 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.data.FruitTransaction;
+import java.util.Map;
+
+public interface OperationHandler {
+    void processWithData(FruitTransaction transaction, Map<String, Integer> data);

```

Comment:
processData will be a more proper name

Reply:
I think OperationHandler and processDate are synonyms, and in this case OperationHandler is more proper name because point directly on its implementations.

Reply:
@BolsunovaNataliia I've meant method naming, `processData` will be enough, no need to use the `with` word here. Anyway this is a minor moment, so nothing critical here